### PR TITLE
Action to automatically create doc issues on PR merge

### PR DESCRIPTION
**Pull request by ianjevans**
This PR creates a GitHub Action workflow that is run when PRs to `master` or `release-*` branches are merged. If the PR's message body has a release note item, and the release note message isn't "None", it will create a corresponding issue in `cockroachdb/docs` with the title of the PR and the release note message.

Currently the docs team runs a script to generate docs release note issues after generating the release notes file for a particular release. That is, issues are created in batches at each alpha/beta/release/patch release. If a release is delayed significantly, the docs team might not become aware of a doc impact issue for weeks.

Creating docs issues when the PR merges allows the docs team to become aware of and triage issues with doc impact in real time.

Release note: None

Related to: #59546

### CODE REVIEW

**Comment by rafiss** [comment]
> If the PR's message body has a release note item, and the release note message isn't "None"

I think it is a fairly common workflow for PR bodies to have no release note, but the commit messages to have them. I think it's because there is other tooling out there that pulls out release notes from the commit messages. Is it possible to inspect the commit message too?

**Comment by ianjevans** [comment]
> > If the PR's message body has a release note item, and the release note message isn't "None"
> 
> I think it is a fairly common workflow for PR bodies to have no release note, but the commit messages to have them. I think it's because there is other tooling out there that pulls out release notes from the commit messages. Is it possible to inspect the commit message too?

Can you find an example of a PR with the release note item in the commit message? I should be able to add a step that searches the github context for comment messages, but it'd be easier if I could look at a real one to figure out where in the event context I should be looking.

File: .github/workflows/doc-issues.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# Action that creates issues in cockroachdb/docs when a PR is merged from
+# cockroachdb/cockroach in the master or release-* branch and has a
+# release note in the PR body.
+
+on:
+  pull_request:
+    branches:
+      - master
+      - 'release-*'
+    types:           [closed]
```
Comment (otan) [inline .github/workflows/doc-issues.yaml:12]: passing comment: is `[closed]` also when someone closes a PR but does not merge it?

Comment (ianjevans) [reply .github/workflows/doc-issues.yaml:12]: The action will trigger in that scenario, yes, but the job has a check to see if the PR was merged on L17.

```
    if:              ${{ github.event.pull_request.merged == true }}
```

GitHub Actions doesn't have a PR merge event, only a close event.

Comment (otan) [reply .github/workflows/doc-issues.yaml:12]: Nice!

**Comment by rafiss** [comment]
@ianjevans here's an example where the PR doesn't have release notes (in this case the commits have `Release note: None`) https://github.com/cockroachdb/cockroach/pull/61713

**Comment by jseldess** [comment]
@ianjevans, this script looks in commit messages for Release note texts, I believe, in case it's helpful: https://github.com/cockroachdb/cockroach/blob/master/scripts/release-notes.py

**Comment by yzdocs** [comment]
Since the approach is substantially different, I have opened another PR for continuation of work here started by @ianjevans. The new PR: https://github.com/cockroachdb/cockroach/pull/65006

**Comment by jlinder** [comment]
@ianjevans, I'm going to close this PR given the goal of the PR is now accomplished by the [Create Docs Issue](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_CreateDocsIssue?mode=builds#all-projects) nightly which creates issues in Jira for any new commits each day.

**Pull request closed without merging**